backend/internal/storage/postgres/profiles.go

```diff
@@ -50,6 +50,73 @@ func (db *DB) GetProfileByUsername(ctx context.Context, username string) (models
 	return profile, nil
 }
 
+func (db *DB) GetProfileByEmail(ctx context.Context, email string) (models.Profile, error) {
```

Comment:
Seems to be a lot of repeated code when comparing `GetProfileByEmail` with `GetProfileById` and the like. I think we could abstract this work into one method.